rust/src/measurements/polars/make_laplace/mod.rs

```diff
@@ -0,0 +1,174 @@
+use crate::core::PrivacyMap;
+use crate::domains::DataTypeFrom;
+use crate::measurements::{get_discretization_consts, make_base_laplace};
+use crate::metrics::AbsoluteDistance;
+use crate::traits::samplers::CastInternalRational;
+use crate::traits::{ExactIntCast, Float, Number};
+use crate::transformations::ToVec;
+use crate::{
+    core::{Function, Measurement, MetricSpace},
+    domains::{AtomDomain, Context, ExprDomain, ExprMetric, VectorDomain},
+    error::Fallible,
+    measures::MaxDivergence,
+    traits::samplers::SampleDiscreteLaplaceZ2k,
+};
+use polars::prelude::*;
+
+/// Polars operator to make the Laplace noise measurement
+///
+/// # Arguments
+/// * `input_domain` - ExprDomain
+/// * `input_metric` - The metric space under which neighboring LazyFrames are compared
+/// * `scale` - Noise scale parameter for the laplace distribution. `scale` == sqrt(2) * standard_deviation.
+pub fn make_laplace_expr<
+    C: Context,
+    MI,
+    TA: Float + CastInternalRational + SampleDiscreteLaplaceZ2k + DataTypeFrom + Send + Sync,
+    QI: Number,
+>(
+    input_domain: ExprDomain<C>,
+    input_metric: MI,
+    scale: TA,
+    k: Option<i32>,
+) -> Fallible<Measurement<ExprDomain<C>, Expr, MI, MaxDivergence<TA>>>
+where
+    (ExprDomain<C>, MI): MetricSpace,
+    MI: ExprMetric<C, InnerMetric = AbsoluteDistance<QI>, Distance = TA>,
+    ChunkedArray<TA::Polars>: ToVec<TA>,
+    i32: ExactIntCast<TA::Bits>,
+    TA::Polars: PolarsNumericType<Native = TA>,
+    Series: NamedFrom<Vec<TA>, [TA]>,
+{
+    if scale.is_sign_negative() {
+        return fallible!(MakeMeasurement, "scale must not be negative");
+    }
+
+    let lap_meas = make_base_laplace::<VectorDomain<AtomDomain<TA>>>(
+        Default::default(),
+        Default::default(),
+        scale.clone(),
+        k,
+    )?;
+
+    let (k, _): (i32, TA) = get_discretization_consts(k)?;
+
+    Measurement::new(
+        input_domain,
+        Function::new_fallible(move |(_frame, expr): &(C::Value, Expr)| -> Fallible<Expr> {
+            Ok(expr.clone().map(
+                move |s: Series| {
+                    let noisy_vec: Vec<TA> = s
+                        .unpack::<TA::Polars>()?
+                        .into_no_null_iter()
+                        .map(|value| TA::sample_discrete_laplace_Z2k(scale.clone(), value, k))
```

Comment:
This PR now sits above the threading PR, so you should be able to use the function from `lap_meas` in here.